AutomaticQC/imosEchoRangeSetQC.m

```diff
@@ -0,0 +1,167 @@
+function [sample_data, varChecked, paramsLog, df] = imosEchoRangeSetQC( sample_data, auto )
+%IMOSECHORANGEQC Quality control procedure for Teledyne Workhorse (and similar)
+% ADCP instrument data, using the echo intensity diagnostic variable.
+%
+% Echo Range test :
+% This test checks the difference between the highest and lowest values in the 4 beams
+% in each bin (echo intensity range, EIR) at each time stamp. 
+% The echo intensity data should ideally be bin-mapped first using
+% adcpBinMappingPP.m routine.
+% If the difference exceeds a threshold, the entire bin is flagged as bad
+%
+% Inputs:
+%   sample_data - struct containing the entire data set and dimension data.
+%   auto - logical, run QC in batch mode
+%
+% Outputs:
+%   sample_data - same as input, with QC flags added for variable/dimension
+%                 data.
+%   varChecked  - cell array of variables' name which have been checked
+%   paramsLog   - string containing details about params' procedure to include in QC log
+%
+% Author:       Guillaume Galibert <guillaume.galibert@utas.edu.au>
+%               Rebecca Cowley <rebecca.cowley@csiro.au>
+%
+
+%
+% Copyright (C) 2017, Australian Ocean Data Network (AODN) and Integrated 
+% Marine Observing System (IMOS).
+%
+% This program is free software: you can redistribute it and/or modify
+% it under the terms of the GNU General Public License as published by
+% the Free Software Foundation version 3 of the License.
+%
+% This program is distributed in the hope that it will be useful,
+% but WITHOUT ANY WARRANTY; without even the implied warranty of
+% MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+% GNU General Public License for more details.
+
+% You should have received a copy of the GNU General Public License
+% along with this program.
+% If not, see <https://www.gnu.org/licenses/gpl-3.0.en.html>.
+%
+narginchk(1, 2);
+if ~isstruct(sample_data), error('sample_data must be a struct'); end
+
+% auto logical in input to enable running under batch processing
+if nargin<2, auto=false; end
+
+varChecked = {};
+paramsLog  = [];
+
+% get all necessary dimensions and variables id in sample_data struct
+idUcur = 0;
+idVcur = 0;
+idWcur = 0;
+idCspd = 0;
+idCdir = 0;
+idABSIC = cell(4, 1);
+for j=1:4
+    idABSIC{j}  = 0;
+end
+lenVar = length(sample_data.variables);
+for i=1:lenVar
+    paramName = sample_data.variables{i}.name;
+    
+    if strncmpi(paramName, 'UCUR', 4),  idUcur = i; end
+    if strncmpi(paramName, 'VCUR', 4),  idVcur = i; end
+    if strcmpi(paramName, 'WCUR'),      idWcur = i; end
+    if strcmpi(paramName, 'CSPD'),      idCspd = i; end
+    if strncmpi(paramName, 'CDIR', 4),  idCdir = i; end
+    for j=1:4
+        cc = int2str(j);
+        if strcmpi(paramName, ['ABSIC' cc]), idABSIC{j} = i; end
+    end
+end
```

Comment:
This for loop and the entire handling of "ids" above and below can be simplified if the getVar function is used.

``` 
compulsory_vars = {'UCUR',...'ABSIC1',...};
for k=1:length(required_vars)
    required_var = compulsory_vars{k};
    if ~getVar(sample_data.variables,required_var)
        return
    end
end
```